docs/machine-learning/tutorials/image-classification-api-transfer-learning.md

```diff
@@ -0,0 +1,451 @@
+> - Use transfer learning to train a custom TensorFlow image classification model
+> - Classify images with the custom model
+
+## Image Classification Transfer Learning Sample Overview
+
+This sample is a .NET Core console application that classifies images using a pre-trained deep learning TensorFlow model. The code for this sample can be found on the [dotnet/machinelearning-samples repository](https://github.com/dotnet/machinelearning-samples) on GitHub.
+
+## Understand the problem
+
+Image classification is a computer vision problem. Image classification takes an image as input and categorizes it into a prescribed class. Some scenarios where image classification is useful include:
+
+- Facial Recognition
+- Emotion detection
+- Medical diagnosis
+- Landmark detection
+
+This tutorial trains a custom image classification model to perform automated visual inspection of bridge decks to identify structures that are damaged by cracks.
+
+## ML.NET Image Classification API
+
+ML.NET provides various ways of performing image classification. This tutorial focuses on the Image Classification API. The Image Classification API makes use of [TensorFlow.NET](https://github.com/SciSharp/TensorFlow.NET), a low-level library that provides C# bindings for the TensorFlow C++ API.
+
+![ML.NET Image Classification API Architecture](./media/image-classification-api-transfer-learning/architecture.png)
+
+### What is transfer learning?
+
+Transfer learning is the process of using knowledge gained while solving one problem and applying it to a different but related problem.
+
+Training a deep learning model from scratch requires setting several parameters, a large amount of labeled training data and a vast amount of compute resources (hundreds of GPU hours). Using a pre-trained model along with transfer learning allows you to shortcut the training process. 
+
+### Training process
+
+The Image Classification API starts the training process by loading a pre-trained TensorFlow model. The training process consists of two steps:
+
+1. Bottleneck phase
+2. Training phase
+
+![Training Steps](./media/image-classification-api-transfer-learning/training.png)
+
+#### Bottleneck phase
+
+During the bottleneck phase, the set of training images are loaded and the values are used as as input, or features, for the frozen layers of the pre-trained model. The frozen layers include all of the layers in the neural network up to the penultimate layer, informally known as the bottleneck layer. These layers are referred to as frozen because no training will occur on these layers and operations are pass-through. It's at these frozen layers that the lower-level patterns that help a model differentiate between the different classes are computed. The larger the number of layers, the more computationally intensive this step is. Fortunately, since this is a one-time calculation, the results can be cached and used in later runs when experimenting with different parameters.
```

Comment:
Although can we be crisper about what exactly is a feature? By this definition the pixels are features?

Reply:
#resolved